### PR TITLE
Add new metrics

### PR DESCRIPTION
**Pull request by sebr72**
Timers to record different entry points

### CODE REVIEW

**Review by sbrunner — APPROVED** [review]
I'm just relay not sure if that's a good idea to separate the no app id!

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java [hunk 1]
```diff
@@ -172,7 +178,11 @@ public final PrintJobResult call() throws Exception {
       if (Thread.currentThread().isInterrupted()) {
         LOGGER.info(
             "Print job canceled {}\n{}", this.entry.getRequestData(), this.entry.getReferenceId());
-        this.metricRegistry.counter(getClass().getName() + ".canceled").inc();
+        this.metricRegistry
+            .counter(
+                MetricRegistry.name(
+                    getClass().getSimpleName(), ".ExceptionCanceledReportGeneration"))
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:184]: extra dot?

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java [hunk 2]
```diff
@@ -237,9 +244,10 @@ private void sendErrorEmail(final SmtpConfig config, final PJsonObject request,
     message.setContent(multipart);
 
     LOGGER.info("Emailing error to {}", to);
-    Timer.Context timer = this.metricRegistry.timer(getClass().getName() + ".email.success").time();
-    Transport.send(message);
-    timer.stop();
+    try (Timer.Context ignored =
+        this.metricRegistry.timer(getClass().getSimpleName() + ".SendingErrorEmail").time()) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:248]: Use MetricRegistry.name?

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java [hunk 3]
```diff
@@ -310,9 +318,10 @@ private void sendEmail(
     message.setContent(multipart);
 
     LOGGER.info("Emailing result to {}", to);
-    Timer.Context timer = this.metricRegistry.timer(getClass().getName() + ".email.error").time();
-    Transport.send(message);
-    timer.stop();
+    try (Timer.Context ignored =
+        this.metricRegistry.timer(getClass().getSimpleName() + ".SendingResultEmail").time()) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:322]: Use MetricRegistry.name?

File: core/src/main/java/org/mapfish/print/servlet/job/impl/ThreadPoolJobManager.java [hunk 4]
```diff
@@ -331,7 +331,9 @@ protected PrintJob createJob(final PrintJobEntry entry) {
   private void submitInternal(final PrintJobEntry jobEntry) {
     final long numberOfWaitingRequests = this.jobQueue.getWaitingJobsCount();
     if (numberOfWaitingRequests >= this.maxNumberOfWaitingJobs) {
-      metricRegistry.counter(getClass().getName() + ".queue_overflow").inc();
+      metricRegistry
+          .counter(MetricRegistry.name(getClass().getSimpleName(), "queue_overflow"))
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/impl/ThreadPoolJobManager.java:335]: camel case?

**Review by sbrunner — APPROVED** [review]

File: core/src/main/java/org/mapfish/print/servlet/job/HibernateAccounting.java [hunk 5]
```diff
@@ -75,14 +76,18 @@ private void insertRecord(final HibernateAccountingEntry tuple) {
...
                 final Session currentSession = HibernateAccounting.this.sf.getCurrentSession();
                 currentSession.merge(tuple);
                 currentSession.flush();
                 currentSession.evict(tuple);
               }
             });
       } catch (HibernateException ex) {
+        String name =
+            MetricRegistry.name(
+                getClass().getSimpleName(), "insertRecordFailedWithHiddenHibernateException");
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/HibernateAccounting.java:89]: Hidden?
It's in the logs!

Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/servlet/job/HibernateAccounting.java:89]: I think that the following one are more representative: `insertRecordFailedSkiped`, or `insertRecordFailedThenSkiped`, not?

File: core/src/main/java/org/mapfish/print/servlet/job/HibernateAccounting.java [hunk 6]
```diff
@@ -75,15 +77,20 @@ private void insertRecord(final HibernateAccountingEntry tuple) {
...
                 currentSession.merge(tuple);
                 currentSession.flush();
                 currentSession.evict(tuple);
               }
             });
       } catch (HibernateException ex) {
-        LOGGER.warn("Cannot save accounting information", ex);
+        String name =
+            MetricRegistry.name(
+                getClass().getSimpleName(), "insertRecordInHibernatrFailedWithException");
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/HibernateAccounting.java:90]: When we read the metrics, we don't want to know that it was an exception!
```suggestion
                getClass().getSimpleName(), "insertRecordInDatabaseFailed");
```

File: core/src/main/java/org/mapfish/print/metrics/ApplicationStatus.java [hunk 7]
```diff
@@ -28,28 +35,57 @@ class ApplicationStatus extends HealthCheck {
...
       } else {
-        return Result.healthy("This server instance is printing" + health);
+        return createResult(Result.healthy("This server instance is printing" + health));
       }
     } else {
       throw notificationForBrokenServer();
     }
   }
 
+  private Result createResult(final Result current) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/metrics/ApplicationStatus.java:61]: I don't understand what's this method for? Honestly, it looks like a hack.

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java [hunk 8]
```diff
@@ -127,7 +128,11 @@ protected abstract PrintJobResult createResult(
   @Override
   public final PrintJobResult call() throws Exception {
     SecurityContextHolder.setContext(this.securityContext);
-    final Timer.Context timer = this.metricRegistry.timer(getClass().getName() + ".call").time();
+    final Timer.Context timer =
+        this.metricRegistry
+            .timer(
+                MetricRegistry.name(getClass().getSimpleName(), "ActualReportGenerationDuration"))
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:134]: I don't know what actual refer to?
```suggestion
                MetricRegistry.name(getClass().getSimpleName(), "ReportGenerationDuration"))
```

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java [hunk 9]
```diff
@@ -172,15 +179,19 @@ public final PrintJobResult call() throws Exception {
       if (Thread.currentThread().isInterrupted()) {
         LOGGER.info(
             "Print job canceled {}\n{}", this.entry.getRequestData(), this.entry.getReferenceId());
-        this.metricRegistry.counter(getClass().getName() + ".canceled").inc();
+        this.metricRegistry
+            .counter(
+                MetricRegistry.name(
+                    getClass().getSimpleName(), "ExceptionCanceledReportGeneration"))
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:185]: What exception for?
```suggestion
                    getClass().getSimpleName(), "ReportGenerationCanceled"))
```

**Review by sbrunner — CHANGES_REQUESTED** [review]
I passed the metric names in review and proposed some changes :-)

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java (hunk 9 above)
Comment (sebr72) [reply core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:185]: Exception in line 178. The goal is to remove ambiguity with org.mapfish.print.servlet.MapPrinterServlet#cancelPath

File: core/src/main/java/org/mapfish/print/metrics/ApplicationStatus.java (hunk 7 above)
Comment (sebr72) [reply core/src/main/java/org/mapfish/print/metrics/ApplicationStatus.java:61]: It creates Unhealthy Result when we have an "unhealthy" counter different from 0

Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/metrics/ApplicationStatus.java:61]: As I remember, it's precisely what we want to remove?

This kind of "unhealthy" counter => no pod restart => no unhealthy

File: core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java (hunk 8 above)
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:134]: It's not I understand by reading the metric name, what about `OnlyReportGenerationDuration`?

Comment (sebr72) [reply core/src/main/java/org/mapfish/print/servlet/job/PrintJob.java:134]: I will go with real instead of actual to avoid french ambiguity

**Pull request merged**